nox/_options.py

```diff
@@ -122,11 +123,15 @@ def _posargs_finalizer(value, args):
     return posargs[dash_index + 1 :]
 
 
-def _session_completer(prefix, parsed_args, **kwargs):
+def _session_completer(
+    prefix: str, parsed_args: argparse.Namespace, **kwargs
+) -> List[str]:
     global_config = parsed_args
     module = load_nox_module(global_config)
     manifest = discover_manifest(module, global_config)
     filtered_manifest = filter_manifest(manifest, global_config)
+    if isinstance(filtered_manifest, int):  # pragma: no cover
```

Comment:
If this is never hit, we should just remove the branch.

Reply:
mypy complains that it might be an int, so it's there as a type refinement since `filter_manifest` can return an `int`. I can look into adding a test case to cover this line.